script.js

```diff
@@ -1,4 +1,265 @@
+var shuffledDeck = shuffleCards(makeDeck());
+
+// var playerDraw = function (){
+
+// var firstCard =
+
+// }
+
+//Card calculator
+var calculateScore = function (drawnDeck) {
+  var score = 0;
+
+  console.log(drawnDeck);
+  score += drawnDeck[0].rank;
+  for (var i = 1; i < drawnDeck.length; i++) {
+    //Convert rank of Ace to 11 if less than 10
+    if (score < 10 && drawnDeck[i].name == "ace") {
+      console.log("ace");
+      drawnDeck[i].rank = 11;
+    }
+    score += drawnDeck[i].rank;
+    console.log(score);
+  }
+
+  return score;
+};
+
+//Ace value
+
+// var checkForAce = function (drawnDeck) {
+//   var score = 0;
+
+//   for (i = 0; i < drawnDeck.length; i++) {
+//     if (score > 10 && drawnDeck[i].name == "ace") {
+//       drawnDeck[i].rank = 1;
+//     }
+//     score += drawnDeck[i].rank;
+//   }
+//   console.log(score);
+//   return score;
+// };
+
+//cardDrawn Statement
+
+var cardDrawnsStatement = function (person, drawnDeck) {
+  var statement = `${person} has drawn`;
+  console.log(statement);
+  for (i = 0; i < drawnDeck.length; i++) {
+    if (i != drawnDeck.length - 1) {
+      statement += ` ${drawnDeck[i].name} of ${drawnDeck[i].suit} and`;
+    } else statement += ` ${drawnDeck[i].name} of ${drawnDeck[i].suit}.`;
+  }
+  console.log(statement);
+  return statement;
+};
+
+//convert Jack, Queen, King to rank 10
+var convertValues = function (drawnDeck) {
+  for (i = 0; i < drawnDeck.length; i++) {
+    if (
+      drawnDeck[i].name == "king" ||
+      drawnDeck[i].name == "queen" ||
+      drawnDeck[i].name == "jack"
+    ) {
+      drawnDeck[i].rank = 10;
+    }
+  }
+  return drawnDeck;
+};
+
+//Sort the rank according to order and put Ace at the back
+var selectionSort = function (drawnCards) {
+  //Sort the score according to DESC order
+
+  for (var i = 0; i < drawnCards.length; i++) {
+    //set min to the current iteration of i
+    var min = i;
+    for (var j = i + 1; j < drawnCards.length; j++) {
+      if (drawnCards[j].rank > drawnCards[min].rank) {
+        min = j;
+      }
+    }
+    var temp = drawnCards[i];
+    drawnCards[i] = drawnCards[min];
+    drawnCards[min] = temp;
+  }
+
+  return drawnCards;
+};
+
+//Global Variables
+var allPlayerCards = [];
+var allComputerCards = [];
+var gameMode = "round1";
+var computerCard = [];
+var computerScore = 0;
+var playerCard = [];
+var playerScore = 0;
+var orderAllComputerCards = [];
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var counter = 0;
+
+  var output = "";
+  var playerStatement = "";
+  var computerStatement = "";
+  if (gameMode == "round1") {
+    while (counter < 2) {
+      playerCard = shuffledDeck.pop();
+      allPlayerCards.push(playerCard);
+      computerCard = shuffledDeck.pop();
+      allComputerCards.push(computerCard);
+      counter++;
+    }
+    // convert Jack, Queen, King to rank 10
+    convertValues(allPlayerCards);
+    convertValues(allComputerCards);
```

Comment:
as the ranks of king queen jack remain the same for the entire blackjack game. assigning the rank at the card deck level would be better.